src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectConfig.cs

```diff
@@ -406,12 +399,136 @@ public void Evaluate(IParameterSet parameters, IVariableCollection rootVariableC
+            }
+            catch (Exception)
+            {
+                //do nothing
+                //TODO: error handling
+            }
+
+            var successfulTasks = tasksToRun.Where(t => t.Task.IsCompleted && t.Task.Result.Success & t.Task.Result.Value != null);
+            foreach (var task in successfulTasks)
+            {
+                variableCollection[task.Symbol.Name] = InferTypeAndConvertLiteral(task.Task.Result.Value!);
+            }
+
+            async Task<(bool Success, string? Value)> GetBindedValueAsync(string configuredBinding)
+            {
+                if (string.IsNullOrWhiteSpace(configuredBinding))
+                {
+                    throw new ArgumentException($"'{nameof(configuredBinding)}' cannot be null or whitespace.", nameof(configuredBinding));
+                }
+
+                string? prefix = null;
+                string binding = configuredBinding;
+                if (configuredBinding.Contains(":"))
+                {
+                    prefix = configuredBinding.Substring(0, configuredBinding.IndexOf(":")).Trim();
+                    binding = configuredBinding.Substring(configuredBinding.IndexOf(":") + 1).Trim();
+                }
+                if (string.IsNullOrWhiteSpace(binding))
+                {
+                    binding = configuredBinding;
+                }
+
+                IEnumerable<IBindSymbolSource>? sourcesToSearch = null;
+                if (string.IsNullOrWhiteSpace(prefix))
+                {
+                    sourcesToSearch = allSources;
+                }
+                else
+                {
+                    sourcesToSearch = allSources.Where(s => s.SourcePrefix?.Equals(prefix, StringComparison.OrdinalIgnoreCase) ?? false);
+                    if (!sourcesToSearch.Any())
+                    {
+                        //if there is no matching sources, then use all the sources with full binding
+                        sourcesToSearch = allSources;
+                        binding = configuredBinding;
+                    }
+                }
+
+                var successfulTasks = await RunEvaluationTasks(sourcesToSearch, binding).ConfigureAwait(false);
+                cancellationToken.ThrowIfCancellationRequested();
+                if (!successfulTasks.Any())
+                {
+                    //if nothing is found, try all sources with unparsed values from configuration
+                    successfulTasks = await RunEvaluationTasks(allSources, configuredBinding).ConfigureAwait(false);
+                }
+                cancellationToken.ThrowIfCancellationRequested();
+
+                if (!successfulTasks.Any())
+                {
+                    return (false, null);
+                }
+                else if (successfulTasks.Count() == 1)
+                {
+                    return (true, successfulTasks.Single().Value);
+                }
+                else
```

Comment:
Should we print warning in this case?
This anyways means a suspicious configuration - usage of variables without prefixes, where clashes exist. And hence possible presence of behavior unexpected by user (variable bound to something else than user expected as they were not aware of clashing binding source with higher priority)

Reply:
Added warning

Reply:
I still do not see it :-)
(the case of multiple sources with distinct priorities. I see only warning for clashing priorities)